### PR TITLE
Simpler query params

### PR DESCRIPTION
**Pull request by ryanvazquez**
#### :heavy_check_mark: Checklist

Removes product filters from query parameters for cost insights plugin. Only group and project parameters are now supported.

Adds schema validation to query params.

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Added or updated documentation
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: plugins/cost-insights/src/hooks/useFilters.tsx [hunk 1]
```diff
@@ -16,114 +16,98 @@
...
   useEffect(() => {
-    const queryString = stringify({ ...pageFilters, products: productFilters });
-    if (queryString === qsRef.current) return;
-    qsRef.current = queryString;
-    // TODO Remove workaround once issue is resolved in react-router
-    // (https://github.com/ReactTraining/react-router/issues/7496)
-    // navigate({ ...location, search: queryString });
-    navigate({ ...location, search: `?${queryString}` });
-  }, [pageFilters, productFilters]); // eslint-disable-line react-hooks/exhaustive-deps
+    function setLocationFromPageFilters(filters: PageFilters) {
```
Comment (timbonicus) [inline plugins/cost-insights/src/hooks/useFilters.tsx:83]: Those function names 😘  So beautiful. This is so easy to follow!

File: plugins/cost-insights/src/utils/history.ts [hunk 2]
```diff
@@ -15,9 +15,44 @@
...
+    ...getDefaultPageFilters(groups),
+    ...(queryParams.project ? { project: queryParams.project } : {}),
+    ...(queryParams.group ? { group: queryParams.group } : {}),
+  };
+};
+
+export const getInitialProductState = (config: ConfigContextProps) =>
+  config.products.map(product => ({
+    productType: product.kind,
+    duration: Duration.P1M,
```
Comment (timbonicus) [inline plugins/cost-insights/src/utils/history.ts:57]: Can you change this to P30D? We should default to current data when possible, as we do internally.

**Review by timbonicus — APPROVED** [review]
Ahh so nice, awesome changes @ryanvazquez! 🙌

**Review by brendasukh — APPROVED** [review]
`http://localhost:3000/cost-insights?group=pied-piper&project=project-c` so much easier on the eyes🔥

**Review by freben — APPROVED** [review]
Oh hey, missed that this needed our approval :)

**Pull request merged**